1.3.0 panic: runtime error: slice bounds out of range [1:0]

<!--
To increase the chance of your issue getting fixed, open an issue FOR EACH problem. Please do not report multiple problems in a single issue, otherwise they'll probably never get ALL fixed.

Please include all sections of this template into your issue, or it will be automatically closed.
-->

## Which version are you using?

v1.3.0

## Which operating system are you using?

<!-- fill checkboxes with a x. Example: [x] Linux -->

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [ ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ x] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue
when publish by chrome: http://127.0.0.1:8889/publish, it will crash.


## Describe how to replicate the issue

<!--
the maintainers must be able to REPLICATE your issue to solve it - therefore, describe in a very detailed way how to replicate it.
-->

1. start the server
2. publish with(http://127.0.0.1:8889/publish)
3. read with chrome

## Did you attach the server logs?
2023/11/13 09:55:17 INF MediaMTX v1.3.0
2023/11/13 09:55:17 INF configuration loaded from D:\Bible\OBS\mediamtx.yml
2023/11/13 09:55:17 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
2023/11/13 09:55:17 INF [RTMP] listener opened on :1935
2023/11/13 09:55:17 INF [HLS] listener opened on :8888
2023/11/13 09:55:17 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
2023/11/13 09:55:17 INF [SRT] listener opened on :8890 (UDP)
panic: runtime error: slice bounds out of range [1:0]

goroutine 58 [running]:
github.com/bluenviron/mediamtx/internal/protocols/httpserv.(*handlerExitOnPanic).ServeHTTP.func1()
        /s/internal/protocols/httpserv/handler_exit_on_panic.go:21 +0x5e
panic({0x18af3a0?, 0xc0002520f0?})
        /usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/bluenviron/mediamtx/internal/core.(*webRTCHTTPServer).onRequest(0xc0001910b0, 0xc0004a8300)
        /s/internal/core/webrtc_http_server.go:352 +0x8f3
github.com/gin-gonic/gin.(*Context).Next(...)
        /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/context.go:174
github.com/gin-gonic/gin.serveError(0xc0004a8300, 0x194, {0x205cb50, 0x12, 0x12})
        /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/gin.go:656 +0x4b
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc000196340, 0xc0004a8300)
        /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/gin.go:649 +0x485
github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc000196340, {0x1a885b0?, 0xc00052e180}, 0xc0004a8100)
        /go/pkg/mod/github.com/gin-gonic/gin@v1.9.1/gin.go:576 +0x1dd
<!--
Server logs are sometimes useful to identify the issue.
If you think this is the case, set the parameter 'logLevel' to 'debug' and attach the server logs.
-->

yes / no

## Did you attach a network dump?

<!--
If the bug arises when using MediaMTX with an external hardware or software, the most helpful content you can provide is a dump of the data exchanged between the server and the target (network dump), that can be generated in this way:
1) Download wireshark (https://www.wireshark.org/)
2) Start capturing on the interface used for exchanging RTSP (if the server and the target software are both installed on your pc, the interface is probably "loopback", otherwise it's the one of your network card)
3) Start the server and replicate the issue
4) Stop capturing, save the result in .pcap format
5) Attach
-->

yes / no

--- a/internal/core/webrtc_http_server.go
+++ b/internal/core/webrtc_http_server.go
@@ -348,7 +348,7 @@ func (s *webRTCHTTPServer) onRequest(ctx *gin.Context) {
 
 		case len(ctx.Request.URL.Path) >= 2:
 			switch {
-			case strings.HasSuffix(ctx.Request.URL.Path, "/publish"):
+			case len(ctx.Request.URL.Path) > len("/publish") && strings.HasSuffix(ctx.Request.URL.Path, "/publish"):
 				s.onPage(ctx, ctx.Request.URL.Path[1:len(ctx.Request.URL.Path)-len("/publish")], true)
 
 			case ctx.Request.URL.Path[len(ctx.Request.URL.Path)-1] != '/':

--- a/internal/core/webrtc_manager_test.go
+++ b/internal/core/webrtc_manager_test.go
@@ -21,13 +21,13 @@ import (
 
 func TestWebRTCPages(t *testing.T) {
 	p, ok := newInstance("paths:\n" +
-		"  stream:\n")
+		"  all:\n")
 	require.Equal(t, true, ok)
 	defer p.Close()
 
 	hc := &http.Client{Transport: &http.Transport{}}
 
-	for _, path := range []string{"/stream", "/stream/publish"} {
+	for _, path := range []string{"/stream", "/stream/publish", "/publish"} {
 		func() {
 			req, err := http.NewRequest(http.MethodGet, "http://localhost:8889"+path, nil)
 			require.NoError(t, err)

